platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/internal/service/scopes/Scope.java

```diff
@@ -17,75 +17,131 @@
+     * <p>{@link Global} services are visible to {@link CrossBuildSession} services and descendant scopes, but not vice versa.
      */
     interface CrossBuildSession extends Global {}
 
     /**
-     * These services are reused across build invocations in a session.
-     *
-     * A build session can be long-lived in a continuous build (where these services would be reused) or short-lived in a
-     * regular, single build.
-     *
-     * <p>{@link UserHome}, {@link CrossBuildSession} and parent scope services are visible to {@link BuildSession} scope services, but not vice versa.</p>
+     * The scope of a build session.
+     * <p>
+     * A <em>build session</em> represents a single invocation of Gradle, for example when you run {@code gradlew build}.
+     * A session runs the build one or more times.
+     * For example, when continuous build is enabled, the session may run the build many times,
+     * but when it is disabled, the session will run the build only once.
+     * <p>
+     * The build session state is managed by the {@code BuildSessionState} class.
+     * An instance is created at the start of a Gradle invocation and discarded at the end of that invocation.
+     * <p>
+     * {@link UserHome}, {@link CrossBuildSession} and their parent services are visible to
+     * {@link BuildSession} services and descendant scopes, but not vice versa.
      */
     interface BuildSession extends UserHome, CrossBuildSession {}
 
     /**
-     * These services are recreated when in continuous build and shared across all nested builds.
-     * They are closed when the build invocation is completed.
-     *
-     * <p>{@link BuildSession} and parent scope services are visible to {@link BuildTree} scope services, but not vice versa.</p>
+     * The scope of a single <em>build execution</em> within a {@link BuildSession build session}.
+     * <p>
+     * <em>Build tree</em> is another name for the build definition ({@code BuildDefinition}),
+     * which corresponds to composite build.
+     * <p>
+     * The <em>build tree state</em> holds the state for the entire build definition for a single build execution within a session.
+     * The build tree state is managed by the {@code BuildTreeState} class.
+     * An instance is created at the start of a build execution and discarded at the end of that execution.
+     * <p>
+     * {@link BuildSession} and its parent services are visible to {@link BuildTree} services and descendant scopes, but not vice versa.
      */
     interface BuildTree extends BuildSession {}
 
     /**
-     * These services are created once per {@code org.gradle.api.initialization.Settings} the beginning of the build invocation
-     * These services are closed at the end of the build invocation.
-     *
-     * <p>{@link BuildTree} and parent scope services are visible to {@link Build} scope services, but not vice versa.</p>
+     * The scope of a single <em>build</em> within a {@link BuildTree build tree}.
```

Comment:
I feel like this should be made more clear (and simple):
- with continuous builds, it's one of the many cycles
- without continuous build, it's basically the same as the build tree

Reply:
>without continuous build, it's basically the same as the build tree

Not true, `BuildTree` can have multiple `Build` regardless. Each included build in the composite is a `Build`, but they all share the same `BuildTree`.